### ISSUE TITLE
The dataloader for the pix2pix?Can you help me?

### ISSUE THREAD

**Issue by tangjilin** [junyanz/pytorch-CycleGAN-and-pix2pix#302]
Hello！The model I use is Pix2Pix.  
What I want to achieve is : given a RGB image A and a label image B , I want to generate a new RGB image C (corresponding to the label B). In other words ,the input is the stack of A and B, the output is C. However ,in the paper, the input and output is the same channel(3 channels RGB). So what should I do to modify the Dataloader or how should I do to apply my form of input to the Pix2Pix model?Thanks in advance!

**Comment by SsnL** [comment]
It depends on how your dataset is structured. But you should change it so that `real_A` is a 6 channel tensor, and `real_B` is a 3 channel. Then set `input_nc = 6` and `output_nc = 3`. `6` is from assuming that your label image is also 3-channeled. It should be 3 + #channel of the label image.

**Comment by phamnam95** [comment]
Can you suggest any way to structure the input if I have two input for set 1 (A and B) and an output for set 2 (C)? I want to generate C from two input A and B.

**Comment by tangjilin** [comment]
thanks，I will try!

**Comment by tangjilin** [comment]
Maybe you can stack A and B to be the input? @phamnam95

**Comment by phamnam95** [comment]
Thanks. I will try stack along the image_channel dimension.

**Comment by gd1925** [comment]
Hi, I had a question, in this case(ie after stacking the input images along C dimension), should we still run the combine_A_and_B.py code?

**Comment by junyanz** [comment]
I think `combine_A_and_B.py` may not work in this case. You need to modify the `aligned_dataset.py`. Here is a [template](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/data/template_dataset.py) on how to write your custom data loader.

**Comment by gd1925** [comment]
Hello Sir, Thank you very much for your response. 
So Sir, I went through your answers to figure out what you mean by modifying the files, but I am not able to understand it. What does it mean to modify the `aligned_dataset.py` code? In one of the other answers, I read that we would have to use the` unaligned_dataset.py`. If I have to write a new data loader, what exactly must I change?  I found a sample here - [(http://places.csail.mit.edu/deepscene/small-projects/pytorch-CycleGAN-and-pix2pix/data/custom_dataset_data_loader.py )] 
I apologize for the barrage of questions. I tried to understand it, but I am still quite confused. 
Thank you for your time and consideration.

**Comment by junyanz** [comment]
If you use pix2pix, you need to use `aligned_dataset.py`. If you use CycleGAN, you need to use `unaligned_dataset.py`. If you want to implement a new data loading function (such as two inputs, and one output), you can look at the data loader [template](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/data/template_dataset.py). The template has detailed instructions on how to implement your own function.

**Comment by gd1925** [comment]
Hello Sir,
Firstly I am extremely sorry for the delayed response. I had thought I would reply after I figure out the next steps. But I haven't been able to do it. Sir, I have some trouble with using the transformation of images. I understand that I have to transform them to tensors and then return the paths to A and B. In my case I have 4-Channel stacked input and 3-Channel output. I tried using `# dataA = ToTensor()(A) # dataB = ToTensor()(B)` and also these - `params = get_params(self.opt, B.size)
dataA = get_transform(self.opt, params, method=Image.BILINEAR, normalize=False)
dataB = get_transform(self.opt, params)` and ` transform_params = get_params(self.opt, A.size)
        A_transform = get_transform(self.opt, transform_params, grayscale=(self.input_nc == 1))
        B_transform = get_transform(self.opt, transform_params, grayscale=(self.output_nc == 1))`. But I run into runtime errors. I was trying to apply the same transformations on both the images. Is my understanding correct?

Once again, I apologize for the delay. I am very thankful for your time and consideration so far.
